Neos.Neos/Classes/Service/LinkingService.php

```diff
@@ -367,7 +369,19 @@ public function createNodeUri(
         $request = $controllerContext->getRequest()->getMainRequest();
         $uriBuilder = clone $controllerContext->getUriBuilder();
         $uriBuilder->setRequest($request);
-        $action = $workspace && $workspace->isPublicWorkspace() && !$hiddenState->isHidden ? 'show' : 'preview';
+
+        if (
+            $request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && in_array($request->getControllerActionName(), ['edit', 'preview'])
+        ) {
+            $action = $request->getControllerActionName();
+            if ($request->hasArgument('editPreviewMode')) {
+                $arguments['editPreviewMode'] = $request->getArgument('editPreviewMode');
+            }
+        } else {
```

Comment:
We duplicate the logic here. Maybe something like this works:
```suggestion
        if (BackendHelper::isEditMode($request) || BackendHelper::isPreviewMode($request)) {
            $action = $request->getControllerActionName();
            if ($request->hasArgument('editPreviewMode')) {
                $arguments['editPreviewMode'] = $request->getArgument('editPreviewMode');
            }
        } else {
```